panel/layout/fade.py

```diff
@@ -0,0 +1,165 @@
+            self.value()
+        """,
+        "drag": """
+            function endDrag() {
+                document.removeEventListener('mouseup', endDrag);
+                document.removeEventListener('mousemove', handleDrag);
+            }
+            function handleDrag(e) {
+                e = e || window.event;
+                e.preventDefault();
+                current = e.clientX
+                start = view.el.getBoundingClientRect().left
+                value = parseInt(((current-start)/ fade_container.clientWidth)*100)
+                data.value = Math.max(0, Math.min(value, 100))
+            }
+            let e = event || window.event;
+            e.preventDefault();
+            document.addEventListener('mouseup', endDrag);
+            document.addEventListener('mousemove', handleDrag);
+        """,
+        "value": """
+            objects = fade_container.getElementsByClassName('outer');
+            const numObjects = objects.length;
+            const sliderValue = data.value / 100; // Scale data.value to range 0-1
+
+            const opacityStep = 1 / (numObjects - 1);
+            const nearestObjectIndex = Math.floor(sliderValue / opacityStep);
+            const remainder = sliderValue % opacityStep;
+
+            let opacity;
+            for (let i = 0; i < numObjects; i++) {
+                if (i === nearestObjectIndex) {
+                    opacity = 1 - remainder / opacityStep;
+                } else if (i === nearestObjectIndex + 1) {
+                    opacity = remainder / opacityStep;
+                } else {
+                    opacity = (nearestObjectIndex < i && i < nearestObjectIndex + 1) ? 1 : 0;
+                }
+                console.log(opacity, i);
+                objects[i].style.opacity = opacity;
+            }
+            self.adjustSlider();
+        """,
+        "slider_width": "self.adjustSlider()",
+        "adjustSlider": """
+            halfWidth = parseInt(data.slider_width/2)
+            slider.style.marginLeft = `calc(${data.value}% - ${halfWidth}px)`
+        """,
+    }
+
+    _stylesheets: ClassVar[List[str]] = [f"{CDN_DIST}css/fade.css"]
+
+    def __init__(self, *objects, **params):
+        if "objects" in params and objects:
+            raise ValueError(
+                "Either supply objects as a positional argument or "
+                "as a keyword argument, not both."
+            )
+        from ..pane.base import panel
+
+        objects = params.pop("objects", objects)
+        if not objects:
+            objects = [None, None]
+        super().__init__(objects=[panel(obj) for obj in objects], **params)
+
+    @param.depends("objects", watch=True, on_init=True)
+    def _update_layout(self):
+        for i, obj in enumerate(self.objects):
+            if i < len(self._objects):
+                self._objects[i] = obj
+            else:
+                self._objects.append(obj)
+
+    def select(self, selector=None):
```

Comment:
Why do we need this method? (Any method added is costly as it needs documentation, maintenance and support on discourse).

Would a user not just set the `objects` to the objects he/ she wants to display?

Reply:
Not sure actually. Lots of this is migrated from `Swipe`
https://github.com/holoviz/panel/blob/main/panel/layout/swipe.py#L129